messaging/WFLY-13793_disable_legacy_amq1_prefix.adoc

```diff
@@ -0,0 +1,126 @@
+
+[ ] QE
+
+* TBD
+
+=== Affected Projects or Components
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+* Add a `enable-amq1-prefix`  boolean attribute to the https://wildscribe.github.io/WildFly/18.0/subsystem/messaging-activemq/external-jms-queue/[/subsystem=messaging-activemq/external-jms-queue resource] to control the enablement of Artemis 1.x prefix for queue defined on a remote Artemis server. Default is `true` (meaning that name retro-compatibility is enabled). For instance an queue named `testQueue`, by setting this attribute to `false`, it effectively allows you to create an external queue without amq1 prefix `jms.queue`. The created queue will be `testQueue`. Otherwise, it will create an external queue with amq1 prefix `jms.queue`. The created queue will be `jms.queue.testQueue`.
+
+
+* Add a `enable-amq1-prefix`  boolean attribute to the https://wildscribe.github.io/WildFly/18.0/subsystem/messaging-activemq/external-jms-topic/[/subsystem=messaging-activemq/external-jms-topic resource] to control the enablement of Artemis 1.x prefix for topic defined on a remote Artemis server. Default is `true` (meaning that name retro-compatibility is enabled). For instance a topic named `testTopic`, by setting this attribute to `false`, it effectively allows you to create an external topic without amq1 prefix `jms.topic`. The created topic will be `testTopic`. Otherwise, it will create an external topic with amq1 prefix `jms.topic`. The created topic will be `jms.topic.testTopic`.
+
+* For a destination (queue or topic) created by annotation `@JMSDestinationDefinition` on a remote Artemis Server, add a `enable-amq1-prefix` boolean property to control the enablement of Artemis 1.x prefix. If the property is not defined, the default is `true` (meaning that name retro-compatibility is enabled).
+
+[source, java]
+----
+@JMSDestinationDefinition(
+    // explicitly mention a resourceAdapter corresponding to a pooled-connection-factory resource to the remote server
+    resourceAdapter = "remote-artemis",
+    name="java:global/env/myQueue2",
+    interfaceName="javax.jms.Queue",
+    destinationName="myQueue2",
+    properties = {
+        "enable-amq1-prefix=false"
+    }
+)
+----
+
+=== Hard Requirements
+
+N/A
+
+=== Nice-to-Have Requirements
+
+N/A
+
+=== Non-Requirements
+
+N/A
+
+== Test Plan
+
+* Enginner WildFly integration test `ExternalJMSDestinationManagementTestCase`
+
+Prepare Wildfly server messaging-activemq subsystem be able to use internal broker as remote one.
+Create queue `myExternalQueue` and `myExternalTopic` with `enable-amq1-prefix` set to `false`. Both created queues name should not contain amq1 prefix `jms.queue` or `jms.topic`.
+Create queue `myExternalLegacyQueue` and `myExternalLegacyTopic` with `enable-amq1-prefix` set to `true`. Both created queue contains amq1 prefix `jms.queue` or `jms.topic`.
+
+
+
+* Enginner WildFly integration test `ExternalJMSDestinationDefinitionMessagingDeploymentTestCase`
```

Comment:
@soul2zimate if not already present, please also add the "negative" version of the test: it should test that when `enable-amq1-prefix` is not present (or set to true) the prefix is added

Reply:
this is covered https://github.com/wildfly/wildfly/pull/14412/files#diff-252482f0c3167ecf201ab5fea058fea3442d25f28154a05b8829d8e885616f87R92

Reply:
@ehsavoie tnks; @soul2zimate can you add this information here? this way I won't ask again on the TP ....

Reply:
> this is covered https://github.com/wildfly/wildfly/pull/14412/files#diff-252482f0c3167ecf201ab5fea058fea3442d25f28154a05b8829d8e885616f87R92

@ehsavoie The link points to `ExternalJMSDestinationManagementTestCase`, I don't think we have the case when `enable-amq1-prefix` is true in `ExternalJMSDestinationDefinitionMessagingDeploymentTestCase`.

Reply:
Maybe you are right, I didn't create a specific deployment for this test case

Reply:
I've added ExternalJMSDestinationDefinitionLegacyPrefixMessagingDeploymentTestCase (https://github.com/wildfly/wildfly/pull/14412/files#diff-c1e6f0113d3eae573c1878dc135c7f63954abe9f6d8e6482c8a631fe22bb469f) to cover this

Reply:
Thank you, I have added this new `ExternalJMSDestinationDefinitionLegacyPrefixMessagingDeploymentTestCase` in the `Test Plan` section.